### DISCUSSION TITLE
emit

### DISCUSSION THREAD

**Discussion by chenlvzhou** [tauri-apps/tauri#9241] [Q&A]
How to use emit outside of the man function.

**Comment by chenlvzhou** [comment]
fn main() {
    let mut app_builder = tauri::Builder::default();

    let app_context = tauri::generate_context!();

    let app = app_builder.build().expect("Failed to build Tauri app");

    app.setup(|app| {
        let id = app.listen_global("event-name", |event| {
            println!("got event-name with payload {:?}", event.payload());
        });

        Ok(())
    }).expect("Failed to setup Tauri app");

    app.run(app_context).expect("Failed to run Tauri app");

    emit_event(&app).expect("Failed to emit event");
}

fn emit_event(app: &tauri::App) -> Result<(), tauri::Error> {
    app.emit_all("event-name", "Tauri is awesome!")?;
    Ok(())
}

**Comment by shadow3aaa** [comment]
See [command](https://tauri.app/v1/guides/features/command). In a command tauri can provide an apphandle/windows as argument. So you can emit an event in it.
I think it's the right way to do this.

**Comment by chenlvzhou** [comment]
resolved
